File: testing/driver/src/main/java/net/corda/testing/driver/node/FlowFatalException.java
package net.corda.testing.driver.node;

import org.jetbrains.annotations.Nullable;

public class FlowFatalException extends RuntimeException {
    public FlowFatalException(@Nullable String message, @Nullable Throwable cause) {
        super(message, cause);
    }

    public FlowFatalException(@Nullable String message) {
        super(message);
    }
}

File: runtime/components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/exceptions/FlowFatalException.kt
package net.corda.flow.pipeline.exceptions

import net.corda.v5.base.exceptions.CordaRuntimeException

/**
 * The [FlowFatalException] is thrown for an unrecoverable error, this exception signals the flow can't continue
 * and should be moved to the DLQ
 */
class FlowFatalException(override val message: String, cause: Throwable? = null) :
    CordaRuntimeException(message, cause)

